### PR TITLE
Add batches dbstore metrics dashboard

### PR DESCRIPTION
**Pull request by eseliger**
image

### CODE REVIEW

**Review by mrnugget — APPROVED** [review]
Nice!

File: enterprise/internal/batches/store/batch_changes.go [hunk 1]
```diff
@@ -129,7 +139,12 @@ func (s *Store) updateBatchChangeQuery(c *btypes.BatchChange) *sqlf.Query {
 }
 
 // DeleteBatchChange deletes the batch change with the given ID.
-func (s *Store) DeleteBatchChange(ctx context.Context, id int64) error {
+func (s *Store) DeleteBatchChange(ctx context.Context, id int64) (err error) {
+	ctx, endObservation := s.operations.deleteBatchChange.With(ctx, &err, observation.Args{LogFields: []log.Field{
+		log.Int("batchChangeID", int(id)),
```
Comment (LawnGnome) [inline enterprise/internal/batches/store/batch_changes.go:144]: In `DeleteBatchSpec`, we call this field `ID`, whereas here it's `batchChangeID`. (I think we also have some inconsistent capitalisation at points too.)

Is there value in trying to make the field names more consistent before we merge this?

**Review by LawnGnome — APPROVED** [review]
LGTM. I wish this was less manual, but that's not your fault!

File: enterprise/internal/batches/store/batch_changes.go (hunk 1 above)
Comment (eseliger) [reply enterprise/internal/batches/store/batch_changes.go:144]: Fixed!

**Comment by courier-new** [comment]
Questions transplanted from the other PR:

1. How does one test that this is working? Are you running prometheus locally and looking for a certain output? I'd also love to get the 30-second overview from you or someone else at some point how to find a specific metric we're tracking in prod. Is that generally through the Grafana dashboards?

	**EDIT:** Now that I see the screenshots here and the dashboard changes you added, I was able to find the [metrics guide](https://docs.sourcegraph.com/admin/observability/metrics). Is that how one typically tests this?

2. I see the signature of `endObservation` is `type FinishFunc func(count float64, args Args)`, but I don't see a comment as to what either argument is intended for. What is the `count` of, and what are the `args` for? It feels strange that we **always** call it with exactly `1` and the empty `observation.Args{}` struct -- can we not set these as defaults for some reason? I literally [can't find any other occurrences](https://k8s.sgdev.org/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+endObservation%5C%28%5B2-9%5D&patternType=regexp) of us calling this with a `count` other than `1`, for example:

	image

	Anyway that's not really a you problem, I was mostly just surprised by how chunky a PR to add "basic metrics" to store methods got, and I know go code can get very verbose, but this stood out to me.

**Pull request merged**